xdelta3-dir-patcher

```diff
@@ -65,55 +449,96 @@ class XDelta3DirPatcher(object):
 
     # TODO: Unit test me
     def copy_attributes(self, src_file, dest_file):
-        if args.debug: print("Copying mode data")
+        if self.args.verbose: print("Copying file metadata:", dest_file)
         copymode(src_file, dest_file)
-
-        if args.debug: print("Copying stat data (sans UID/GID")
         copystat(src_file, dest_file)
 
-        if args.debug: print("Copying UID & GID")
         uid = stat(src_file).st_uid
         gid = stat(src_file).st_gid
 
         chown(dest_file,uid,gid)
 
-    def _find_file_delta(self, rel_path, new_file, old_root, new_root, target_root):
-        print("Processing %s" % new_file)
+    @staticmethod
+    def _get_temp_dir(staging_dir, prefix):
+        temp_dir = None
+        if staging_dir:
+            folder_name = '%s_%s' % (prefix, XDelta3DirPatcher.get_random_string())
```

Comment:
@sgnn7 Don't do this, just tell mkdtemp the directory to create in using the `dir` parameter.
```
>>> import tempfile
>>> import os
>>> staging_dir = None
>>> tempfile.mkdtemp(prefix='blah', dir=staging_dir)
'/tmp/blahqpetff'
>>> os.rmdir('/tmp/blahqpetff')
>>> staging_dir = '/var/tmp'
>>> tempfile.mkdtemp(prefix='blah', dir=staging_dir)
'/var/tmp/blah9r7fhp'
>>> os.rmdir('/var/tmp/blah9r7fhp')
```
https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp